tests/subsys/bluetooth/mesh/scheduler_model/message_validity/src/main.c

```diff
@@ -0,0 +1,477 @@
+	zassert_not_null(_bt_mesh_scheduler_cli_cb.reset, "Client reset cb is null");
+
+	_bt_mesh_scheduler_srv_cb.reset(sched_mod1_elem);
+	_bt_mesh_scheduler_cli_cb.reset(sched_mod2_elem);
+}
+
+ZTEST_SUITE(scheduler_message_validity_test, NULL, setup, run_before, NULL, teardown);
+
+ZTEST(scheduler_message_validity_test, test_cli_set_prohibited)
+{
+	/** (1) Send a SET and SET_UNACK message with a prohibited schedule entry, one for each
+	 * possible prohibited value according to the spec.
+	 * and (2) verify that the client returns with error (no message is sent to server).
+	 */
+
+	struct bt_mesh_schedule_entry test_entry[6];
+
+	for (size_t i = 0; i < ARRAY_SIZE(test_entry); i++) {
+		test_entry[i] = valid_entry;
+	}
+
+	/* prohibited struct member values */
+	test_entry[0].action = 4;
+	test_entry[1].year = 0x65;
+	test_entry[2].day = 0x21;
+	test_entry[3].hour = 0x1a;
+	test_entry[4].minute = 0x40;
+	test_entry[5].second = 0x40;
+
+	for (int i = 0; i < ARRAY_SIZE(test_entry); i++) {
+		cli_set_msg_send_assert(&test_entry[i], test_idx, -EINVAL);
+	}
+}
+
+ZTEST(scheduler_message_validity_test, test_cli_set_valid)
+{
+	/** With a valid test action entry:
+	 * (1) Verify that the client accepts the valid entry and proceeds to send the set-message(s),
+	 * both acknowledged and unacknowledged.
+	 */
+
+	cli_set_msg_send_assert(&valid_entry, test_idx, 0);
+}
+
+ZTEST(scheduler_message_validity_test, test_cli_get_action_prohibited)
+{
+	/** (1) Send a ACTION GET message with a prohibited schedule action index count.
+	 *  (2) Verify that the client deny the prohibited index value (no message is sent)
+	 */
+
+	test_idx = BT_MESH_SCHEDULER_ACTION_ENTRY_COUNT;
+	zassert_equal(bt_mesh_scheduler_cli_action_get(&sched_cli, NULL, test_idx, NULL), -EINVAL);
+}
+
+ZTEST(scheduler_message_validity_test, test_cli_get_action_valid)
+{
+	/** (1) Send a ACTION GET message with a valid schedule action index count.
+	 *  (2) Verify that the client accepts the valid index value and proceeds to
+	 * successfully send a GET-message.
+	 *  (3) Verify that the client is able to successfully handle a ACTION STATUS response.
+	 */
+
+	test_idx = 1;
+	zassert_ok(bt_mesh_scheduler_cli_action_get(&sched_cli, NULL, test_idx, NULL));
+
+	/* create a dummy packed action */
+	NET_BUF_SIMPLE_DEFINE(dummy_packed, 10);
+	net_buf_simple_add_le32(&dummy_packed, 1);
+	net_buf_simple_add_le32(&dummy_packed, 2);
+	net_buf_simple_add_le16(&dummy_packed, 3);
+
+	zassert_ok(_bt_mesh_scheduler_cli_op[1].func(sched_mod2_elem, &test_ctx, &dummy_packed));
+	zassert_ok(k_sem_take(&action_status_rcvd, K_SECONDS(TIMEOUT)),
+		   "Client failed to call action status handler");
+}
+
+ZTEST(scheduler_message_validity_test, test_cli_get_valid)
+{
+	/** (1) Send a GET message and ensure that the client succeeds in sending the message.
+	 *  (2) Verify that the client is able to successfully handle a STATUS response.
+	 */
+
+	zassert_ok(bt_mesh_scheduler_cli_get(&sched_cli, NULL, NULL));
```

Comment:
This does not actually check that it sends the message. Would probably be nice to do a expect on the mocked message sending function here. Same for the other "check that client can send message X"-situations.